queries/cdmq/add-run.js

```diff
@@ -92,51 +115,106 @@ async function processDir(instance, dir, mode) {
     } catch (error) {
       console.error('Error processing NDJSON.XZ file:', error);
     }
-  }
+    const usage = process.memoryUsage();
+    debuglog('Memory Breakdown after processing ' + filePath);
+    debuglog(`RSS (Resident Set Size): ${Math.round(usage.rss / 1024 / 1024)} MB`);
+    debuglog(`Heap Total: ${Math.round(usage.heapTotal / 1024 / 1024)} MB`);
+    debuglog(`Heap Used: ${Math.round(usage.heapUsed / 1024 / 1024)} MB`);
+    debuglog(`External: ${Math.round(usage.external / 1024 / 1024)} MB`);
+    debuglog(`Array Buffers: ${Math.round(usage.arrayBuffers / 1024 / 1024)} MB`);
 
-  if (mode == 'index') {
-    var responses = await esJsonArrRequest(instance, '', '/_bulk', jsonArr);
-  } else if (mode == 'getinfo') {
-    var info = { runIds: [], indices: {} };
-    console.log('going to process ' + jsonArr.length + ' lines');
-    for (var k = 1; k < jsonArr.length; k += 2) {
-      try {
-        var action = JSON.parse(jsonArr[k - 1]);
-        var doc = JSON.parse(jsonArr[k]);
-      } catch (jsonError) {
-        console.log('Could not porse: [' + jsonArr[k] + ']');
-        continue;
-      }
-      runId = doc['run']['run-uuid'];
-      if (!info['runIds'].includes(runId)) {
-        info['runIds'].push(runId);
-      }
-      // { "index": { "_index": "cdmv8dev-metric_data" } }
-      if (Object.keys(action).includes('index') && Object.keys(action['index']).includes('_index')) {
-        const indexName = action['index']['_index'];
-        const regExp = /^cdm-*(v7dev|v8dev|v9dev)-([^@]+)(@\d\d\d\d\.\d\d)*$/;
-        var matches = regExp.exec(indexName);
-        if (matches) {
-          cdmVer = matches[1];
-          if (!Object.keys(info['indices']).includes(cdmVer)) {
-            info['indices'][cdmVer] = [];
+    // After so much data is accumulated or at the last file, index what we have then clear the array.
+    if (jsonArr.length > 4000000 || i == xzFiles.length) {
```

Comment:
magic number?

Reply:
Same here, a number I used after experimenting with many numbers to keep memory usage not huge while maintaining good throughput.